Azure.ResourceManager.Core/Resources/LocationContainer.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed under the MIT License.
+
+using Azure.ResourceManager.Core.Adapters;
+using Azure.ResourceManager.Resources;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// Represents an Azure geography region where supported resource providers live.
+    /// </summary>
+    public class LocationContainer : OperationsBase
+    {
+        internal LocationContainer(SubscriptionOperations subscriptionOperations)
+            : base(subscriptionOperations.ClientOptions, subscriptionOperations.Id, subscriptionOperations.Credential, subscriptionOperations.BaseUri)
+        {
+        }
+
+        protected override ResourceType ValidResourceType => SubscriptionOperations.ResourceType;
+
+        /// <summary>
+        /// Gets the subscription client.
+        /// </summary>
+        private SubscriptionsOperations SubscriptionsClient => ResourcesClient.Subscriptions;
+
+        /// <summary>
+        /// Gets the Azure subscriptions.
+        /// </summary>
+        /// <returns> Subscription container. </returns>
+        public SubscriptionContainer GetSubscriptionContainer()
+        {
+            return new SubscriptionContainer(ClientOptions, Credential, BaseUri);
+        }
+
+        /// <summary>
+        /// Lists all geo-locations.
+        /// </summary>
+        /// <returns> A collection of location data that may take multiple service requests to iterate over. </returns>
+        /// <exception cref="InvalidOperationException"> <paramref name="subscriptionId"/> is null. </exception>
+        public Pageable<LocationData> List()
+        {
+            if (string.IsNullOrWhiteSpace(Id.Subscription))
+            {
+                throw new InvalidOperationException("Please select a valid subscription");
+            }
+
+            return new PhWrappingPageable<Azure.ResourceManager.Resources.Models.Location, LocationData>(SubscriptionsClient.ListLocations(Id.Subscription), s => s.DisplayName);
+        }
+
+        /// <summary>
+        /// Lists all geo-locations.
+        /// </summary>
+        /// <param name="subscriptionId"> The Id of the target subscription. </param>
+        /// <param name="token"> A token to allow the caller to cancel the call to the service. The default value is <see cref="P:System.Threading.CancellationToken.None" />. </param>
+        /// <returns> An async collection of location data that may take multiple service requests to iterate over. </returns>
+        /// <exception cref="InvalidOperationException"> <paramref name="subscriptionId"/> is null. </exception>
+        public AsyncPageable<LocationData> ListAsync(string subscriptionId = null, CancellationToken token = default(CancellationToken))
+        {
+            async Task<AsyncPageable<LocationData>> PageableFunc()
+            {
+                if (string.IsNullOrWhiteSpace(Id.Subscription))
```

Comment:
this cannot be null due to construction requirements, you can safely remove this check